tools/buildsteps/windows/patches/0005-ffmpeg-windows-dxva2-check-nullptr-surface.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/libavcodec/dxva2.c b/libavcodec/dxva2.c
+--- a/libavcodec/dxva2.c
++++ b/libavcodec/dxva2.c
+@@ -777,7 +777,7 @@ unsigned ff_dxva2_get_surface_index(const AVCodecContext *avctx,
+ #if CONFIG_D3D11VA
+     if (avctx->pix_fmt == AV_PIX_FMT_D3D11)
+         return (intptr_t)frame->data[1];
+-    if (avctx->pix_fmt == AV_PIX_FMT_D3D11VA_VLD) {
++    if (avctx->pix_fmt == AV_PIX_FMT_D3D11VA_VLD && surface) {
+         D3D11_VIDEO_DECODER_OUTPUT_VIEW_DESC viewDesc;
+         ID3D11VideoDecoderOutputView_GetDesc((ID3D11VideoDecoderOutputView*) surface, &viewDesc);
+         return viewDesc.Texture2D.ArraySlice;
+@@ -785,7 +785,7 @@ unsigned ff_dxva2_get_surface_index(const AVCodecContext *avctx,
+ #endif
+ #if CONFIG_DXVA2
+     for (i = 0; i < DXVA_CONTEXT_COUNT(avctx, ctx); i++) {
+-        if (avctx->pix_fmt == AV_PIX_FMT_DXVA2_VLD && ctx->dxva2.surface[i] == surface)
++        if (avctx->pix_fmt == AV_PIX_FMT_DXVA2_VLD && surface && ctx->dxva2.surface[i] == surface)
```

Comment:
I think this is unnecessary. The current code already works fine for surface null because it won't ever match a ctx->dxva2.surface[i]

The orig patch had a change inside #if CONFIG_DXVA2 for an early exit optimization I guess, but it was not for bug-fixing. Also CONFIG_DXVA2 / AV_PIX_FMT_DXVA2_VLD are for D3D9, which Kodi doesn't use anymore.

Reply:
Removed this part of patch as this code path is not used anyway